### PR TITLE
footswitch: init at unstable-2021-03-17

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

Footswitch is a tool to configure HID foot pedals

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/inputmethods/footswitch/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "footswitch";
+  version = "1.0+20210317.aa0b10f";
+
+  meta = with lib; {
+    description = "Command line utlities for programming PCsensor and Scythe foot switches.";
+    homepage    = "https://github.com/rgerganov/footswitch";
+    license     = licenses.mit;
+    platforms   = platforms.linux;
+    maintainers = with maintainers; [ baloo ];
+  };
```
Comment (sbruder) [inline pkgs/tools/inputmethods/footswitch/default.nix:13]: I don’t know if there is an official guideline for that, but it’s conventional to have the meta information at the end of the file (cf. [nixpkgs-hammering](https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md))

File: pkgs/tools/inputmethods/footswitch/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, pkg-config, hidapi }:
+
+stdenv.mkDerivation {
+  pname = "footswitch";
+  version = "1.0+20210317.aa0b10f";
```
Comment (sbruder) [inline pkgs/tools/inputmethods/footswitch/default.nix:5]: The [nixpkgs manual requires](https://nixos.org/manual/nixpkgs/stable/#sec-package-naming) the usage of the `unstable-YYYY-MM-DD` format for packaging unreleased software.

```suggestion
  version = "unstable-2021-03-17";
```

**Review by sbruder — CHANGES_REQUESTED** [review]
I left a suggestion about attribute ordering and the version format. Also, the commit message should be formatted `footswitch: init at unstable-2021-03-17` (cf. [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/d3841d1e7bcfe061800816f1b414288bc04b4663/CONTRIBUTING.md#submitting-changes)).

Otherwise the package builds and executing the binary shows the usage help, though I don’t have the hardware to test its functionality.

**Comment by baloo** [comment]
Thanks for the review!

**Review by sbruder — APPROVED** [review]

File: pkgs/tools/inputmethods/footswitch/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "rgerganov";
+    repo = "footswitch";
+    rev = "aa0b10f00d3e76dac27b55b88c8d44c0c406f7f0";
+    sha256 = "sha256-SikYiBN7jbH5I1x5wPCF+buwFp1dt35cVxAN6lWkTN0=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ hidapi ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/footswitch/default.nix:17]:
```suggestion
  postPatch = ''
```

**Pull request merged**